src/test/java/uk/gov/pay/directdebit/webhook/gocardless/services/handlers/GoCardlessHandlerTest.java

```diff
@@ -41,43 +38,24 @@
     @Spy
     GoCardlessEvent goCardlessEvent = GoCardlessEventFixture.aGoCardlessEventFixture().toEntity();
 
-    PaymentRequestEvent paymentRequestEvent = PaymentRequestEventFixture.aPaymentRequestEventFixture().toEntity();
+    Event event = EventFixture.aPaymentRequestEventFixture().toEntity();
 
     @Test
     public void shouldLinkToPaymentRequestEventAndStoreEventIfActionIsHandled() {
-        goCardlessHandler = new GoCardlessHandler(mockedTransactionService, mockedPayerService, mockedGoCardlessService) {
-            @Override
-            protected Map<GoCardlessAction, Function<Transaction, PaymentRequestEvent>> getHandledActions() {
+        goCardlessHandler = new GoCardlessHandler(mockedTransactionService, mockedGoCardlessService) {
+            protected Map<GoCardlessAction, Function<Transaction, Event>> getHandledActions() {
                 return ImmutableMap.of(
                         GoCardlessPaymentHandler.GoCardlessPaymentAction.PAID_OUT,
-                        transaction -> paymentRequestEvent);
-            }
-
-            @Override
-            protected Optional<PaymentRequestEvent> process(GoCardlessEvent event) {
-                return Optional.of(paymentRequestEvent);
-            }
-        };
-        goCardlessHandler.handle(goCardlessEvent);
-        verify(goCardlessEvent).setPaymentRequestEventId(paymentRequestEvent.getId());
-        verify(mockedGoCardlessService).storeEvent(goCardlessEvent);
-    }
-
-    @Test
-    public void shouldNotLinkToPaymentRequestEventButStillStoreEventIfActionIsNotHandled() {
```

Comment:
Another one — did we make a decision to stop storing unhandled GoCardless events?

Reply:
because storing events at this level was unnecessary as it has been already stored in WebhookGoCardlessService (line 33). Too lazy to copy the link.